collect_app/src/main/java/org/odk/collect/android/activities/GeoPolyActivity.java

```diff
@@ -111,6 +112,8 @@
     private Double restoredMapZoom;
     private List<MapPoint> restoredPoints;
 
+    private static boolean isSettingsDialogActive;
```

Comment:
We shouldn't use static state like this. If one instance of `GeoPolyActivity` set this to `true` and then another was opened later it would behave as if the dialog was open. We could manage all this with extra logic but it'd be a headache. I think you should look at a solution using a `DialogFragment` like you did for the quit form dialog.

I'll convert this PR to a draft so you can make changes in peace.